app/src/main/java/org/httpsknighthacks/knighthacksandroid/MainActivity.java

```diff
@@ -67,31 +71,29 @@ protected void onCreate(Bundle savedInstanceState) {
             manager.createNotificationChannel(channel);
         }
 
+        subscribeToTopic();
+
+        // End of firebase code.
+
         setContentView(R.layout.activity_main);
 
         mRequestQueueSingleton = new RequestQueueSingleton(getApplicationContext());
         getImageAndTitles();
+    }
 
-        // Shows the Token to the screen
-
-        /*FirebaseInstanceId.getInstance().getInstanceId()
-                .addOnCompleteListener(new OnCompleteListener<InstanceIdResult>() {
+    private void subscribeToTopic() {
+        FirebaseMessaging.getInstance().subscribeToTopic("general")
+                .addOnCompleteListener(new OnCompleteListener<Void>() {
                     @Override
-                    public void onComplete(@NonNull Task<InstanceIdResult> task) {
+                    public void onComplete(@NonNull Task<Void> task) {
+                        String msg = "Successful: Subscribed to General";
```

Comment:
If you are going to reuse the term for the topic, you may want to create a constant for it and declare that at the beginning of the file or something like that.